packages/js/k6/rollup.config.ts

```diff
@@ -18,7 +20,7 @@ function getConfig(testFile: string) {
       preserveModules: true,
       preserveModulesRoot: "src",
     },
-    plugins: [typescript()],
+    plugins: [typescript(), nodeResolve(), commonjs()],
```

Comment:
New rollup plugins used to support bundling npm dependencies, because k6 doesn't support them unless they are bundled in. In our case, we need it for `core-js` to provide a sensible URL API.

I chose this as an alternative to the jslib URL implementation because (a) that still requires importing `URL` and (b) there are no means to support types for it. The suggested approach for supporting types from k6 jslib libraries is to vendor the library code.

I did not try declaring the types using `declare module "...jslib..."` so I'll try that for good measure (which would be nice for the other jslib function we use).

_However_, I've found that so far the k6 and jslib implementations of certain Web APIs have very subtle and annoying differences to the specification, which mostly just causes a bit of mental overload trying to keep track of. Which is to say, it's nice to be able to bundle in polyfills from core-js that will behave according to the specification!

Reply:
k6-jslib-url just exports core-js: https://github.com/grafana/k6-jslib-url/blob/main/index.src.js

For the sake of making sure bundling in node dependencies is sorted for anyone else working in these tests, I think it's best to go ahead and use core-js directly in this PR, as a means of integrating the bundling process before it's needed otherwise.